validator/coordinator/src/main/java/tech/pegasys/teku/validator/coordinator/ValidatorApiHandler.java

```diff
@@ -325,6 +346,16 @@ private boolean headBlockIsTooFarBehind() {
     return headEpoch.plus(1).isLessThan(currentEpoch);
   }
 
+  private List<ValidatorDuties> getValidatorDutiesFromIndexes(
+      final BeaconState state, final UInt64 epoch, final Collection<Integer> validatorIndexes) {
+    List<BLSPublicKey> publicKeys =
+        validatorIndexes.stream()
+            .map(index -> state.getValidators().get(index))
+            .map(Validator::getPubkey)
+            .collect(toList());
+    return getValidatorDutiesFromState(state, epoch, publicKeys);
+  }
```

Comment:
This method seems to be converting from index, to public key, then a couple of levels down `getValidatorDutiesFromState` converts from public key back to index.

Given we don't need the proposal slots it could just call `createValidatorDuties` directly and pass an empty map for the proposal slots. Ideally though we'd split the internal representation of `ValidatorDuties` in two so internally we calculate attestations and proposals separately, then for the existing API we combine them at the API layer.  Currently we're calculating proposer duties when the get attestations endpoint is called.